connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerErrantRecordReporterTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.connect.runtime;
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.apache.kafka.connect.storage.Converter;
+import org.apache.kafka.connect.storage.HeaderConverter;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore("javax.management.*")
+public class WorkerErrantRecordReporterTest {
+
+    private WorkerErrantRecordReporter reporter;
+
+    private KafkaProducer<byte[], byte[]> producer = mock(KafkaProducer.class);
+    private SinkConnectorConfig sinkConnectorConfig = mock(SinkConnectorConfig.class);
+    private Converter converter = mock(Converter.class);
+    private HeaderConverter headerConverter = mock(HeaderConverter.class);
+    private SinkRecord record = mock(SinkRecord.class);
+
+    @Before
+    public void setup() {
+      reporter = new WorkerErrantRecordReporter(
+            producer,
+            sinkConnectorConfig,
+            converter,
+            converter,
+            headerConverter
+        );
+    }
+
+    @Test
+    public void testReport() {
+      when(sinkConnectorConfig.dlqTopicName()).thenReturn("dlq-topic");
+      when(sinkConnectorConfig.enableErrorLog()).thenReturn(false);
+      reporter.report(record, new Throwable());
```

Comment:
maybe make this into a mock as well and at least assert on some method calls using `verify()`

Reply:
I've changed the test. Please check it out again.